openstudiocore/src/utilities/filetypes/EpwFile.cpp

```diff
@@ -480,72 +532,98 @@ static double psat(double T)
-    return boost::optional<std::string>();
+  if(!value) {
+    LOG_FREE(Error,"openstudio.EpwFile", "Missing atmospheric station pressure on " << date << " at " << hms);
+    return boost::none;
   }
   double p = value.get();
-  output << m_atmosphericStationPressure;
-  if(!windSpeed())
-  {
-    LOG_FREE(Error,"openstudio.EpwFile",QString("Missing wind speed on %1 at %2").arg(date).arg(hms).toStdString());
-    return boost::optional<std::string>();
+  string += '\t' + m_atmosphericStationPressure;
+  if(!windSpeed()) {
+    LOG_FREE(Error,"openstudio.EpwFile", "Missing wind speed on " << date << " at " << hms);
+    return boost::none;
   }
-  output << m_windSpeed;
-  if(!windDirection())
-  {
-    LOG_FREE(Error,"openstudio.EpwFile",QString("Missing wind direction on %1 at %2").arg(date).arg(hms).toStdString());
-    return boost::optional<std::string>();
+  string += '\t' + m_windSpeed;
+  if(!windDirection()) {
+    LOG_FREE(Error,"openstudio.EpwFile", "Missing wind direction on " << date << " at " << hms);
+    return boost::none;
   }
-  output << m_windDirection;
+  string += '\t' + m_windDirection;
   double pw;
   value = relativeHumidity();
-  if(!value) // Don't have relative humidity - this has not been tested
-  {
+  if(!value) { // Don't have relative humidity - this has not been tested
     value = dewPointTemperature();
-    if(!value)
-    {
-      LOG_FREE(Error,"openstudio.EpwFile",QString("Cannot compute humidity ratio on %1 at %2").arg(date).arg(hms).toStdString());
-      return boost::optional<std::string>();
+    if(!value) {
+      LOG_FREE(Error,"openstudio.EpwFile", "Cannot compute humidity ratio on " << date << " at " << hms);
+      return boost::none;
     }
     boost::optional<double> optdewpoint = openstudio::convert(value.get(), "C", "K");
     OS_ASSERT(optdewpoint);
     double dewpoint = optdewpoint.get();
     pw = psat(dewpoint);
-  }
-  else // Have relative humidity
-  {
+  } else  { // Have relative humidity
     double pws = psat(drybulb);
     pw = 0.01*value.get()*pws;
   }
   double W = 0.621945*pw/(p-pw);
-  output << QString("%1").arg(W*1000); // need g/kg
+  string += "\t" + std::to_string(W*1000); // need g/kg
   // Pass on solar flux quantities
-  output << "0" << "0";
+  string +=  "\t0\t0";
   // Pass on Tsky
-  output << "0";
+  string += "\t0";
   // Pass on snow and rain
-  output << "0" << "0";
-  return boost::optional<std::string>(output.join("\t").toStdString());
+  string += "\t0\t0";
+  return boost::optional<std::string>(string);
+}
+
+// Local convenience functions
+static int stringToInteger(const std::string &string, bool *ok)
+{
+  int value = 0;
+  *ok = true;
+  try {
+    value = std::stoi(string);
+  }
+  catch (const std::invalid_argument) {
+    *ok = false;
+  }
+  catch (const std::out_of_range) {
+    *ok = false;
+  }
+  return value;
+}
+
+static double stringToDouble(const std::string &string, bool *ok)

```

Comment:
My question was why make it static?  I ask out of curiousity, not because I know better.  Does this prevent it from conflicting with other functions of the same name or something?

http://stackoverflow.com/questions/7610668/are-there-any-advantages-to-making-a-free-static-function